### ISSUE TITLE
Access restrictions

### ISSUE THREAD

**Issue by EvgLol** [isuru89/moodle-reminders-for-calendar-events#79]
Hello.
Thanks for this useful module.
Are the access restrictions of a particular activity considered while sending reminders?

F.e. I create a quiz which will be visible (accessible) to students only when certain conditions are satisfied (F.e. completion of a previous level quiz).

It seems to me, that none of the completion criteria are checked before sending a reminder message.
Student gets a reminder message with a link. Clicking on the link leads him to a course module which is hidden by access restrictions.
I did not find any functions which check availability for the user in local_reminders:
- get_fast_modinfo($this->course, $userid);
- get_availability_tree();
- $tree->check_available(false, $this, $grabthelot, $userid);

Am I wrong or missing something out?
Is this feature planned?

**Comment by isuru89** [comment]
@EvgLol 
No, you are correct. I have added a completion check for the assignment with the function provided by mod_assign module. [see here](https://github.com/isuru89/moodle-reminders-for-calendar-events/blob/master/contents/activity_handlers.class.php#L224).

However, this has not been done for the quiz. It checks only for the given user has necessary permissions for the attempt. [see here](https://github.com/isuru89/moodle-reminders-for-calendar-events/blob/master/contents/activity_handlers.class.php#L159) 

Your PR is welcome, if you have an idea about those functions.
